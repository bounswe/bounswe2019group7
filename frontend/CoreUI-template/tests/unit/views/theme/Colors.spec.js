import Vue from 'vue'
import { shallowMount, mount } from '@vue/test-utils'
import BootstrapVue from 'bootstrap-vue'
import Colors from '@/views/theme/Colors'

Vue.use(BootstrapVue)

describe('Colors.vue', () => {
  it('has a name', () => {
    expect(Colors.name).toMatch('colors')
  })
  it('is Vue instance', () => {
    const wrapper = shallowMount(Colors)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Colors', () => {
    const wrapper = shallowMount(Colors)
    expect(wrapper.is(Colors)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Colors)
    expect(wrapper.find('header.card-header > div').text()).toMatch('Theme colors')
  })
  test('renders correctly', () => {
    const wrapper = mount(Colors)
    expect(wrapper.element).toMatchSnapshot()
  })
})
