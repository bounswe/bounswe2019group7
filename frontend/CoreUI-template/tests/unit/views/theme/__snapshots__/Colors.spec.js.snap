// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors.vue renders correctly 1`] = `
<div
  class="animated fadeIn"
>
  <div
    class="card"
  >
    <!---->
    <header
      class="card-header"
    >
      <div>
        <i
          class="icon-drop"
        />
         Theme colors
    
      </div>
    </header>
    <div
      class="card-body"
    >
      <!---->
      <!---->
       
      <div
        class="card-body"
      >
        <!---->
        <!---->
        <div
          class="row"
        >
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-primary"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Primary Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-secondary"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Secondary Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-success"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Success Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-danger"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Danger Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-warning"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Warning Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-info"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Info Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-light"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Light Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-dark"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Dark Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
   
  <div
    class="card"
  >
    <!---->
    <header
      class="card-header"
    >
      <div>
        <i
          class="icon-drop"
        />
         Grays
    
      </div>
    </header>
    <div
      class="card-body"
    >
      <!---->
      <!---->
       
      <div
        class="card-body"
      >
        <!---->
        <!---->
        <div
          class="row"
        >
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-100"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 100 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-200"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 200 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-300"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 300 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-400"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 400 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-500"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 500 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-600"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 600 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-700"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 700 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-800"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 800 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-gray-900"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand 900 Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
   
  <div
    class="card"
  >
    <!---->
    <header
      class="card-header"
    >
      <div>
        <i
          class="icon-drop"
        />
         Additional colors
    
      </div>
    </header>
    <div
      class="card-body"
    >
      <!---->
      <!---->
       
      <div
        class="card-body"
      >
        <!---->
        <!---->
        <div
          class="row"
        >
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-blue"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Blue Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-indigo"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Indigo Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-purple"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Purple Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-pink"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Pink Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-red"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Red Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-orange"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Orange Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-yellow"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Yellow Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-green"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Green Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-teal"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Teal Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
           
          <div
            class="mb-4 col-sm-6 col-md-4 col-xl-2"
            xs="12"
          >
            <div
              class="theme-color w-75 rounded mb-3 bg-cyan"
              style="padding-top: 75%;"
            />
             
            <h6>
              Brand Cyan Color
            </h6>
             
            <table
              class="w-100"
            >
              <tbody>
                <tr>
                  <td
                    class="text-muted"
                  >
                    HEX:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    #ffffff
                  </td>
                </tr>
                 
                <tr>
                  <td
                    class="text-muted"
                  >
                    RGB:
                  </td>
                   
                  <td
                    class="font-weight-bold"
                  >
                    rgb(255, 255, 255)
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </div>
</div>
`;
